src/Raven.Server/Storage/Schema/Updates/Documents/60000/From50002.cs

```diff
@@ -0,0 +1,123 @@
+
+            InsertIndexValuesFor(step, ConflictsSchemaBase60, ConflictsBucketAndEtagSlice,
+                fixedSizeIndex: ConflictsSchemaBase60.FixedSizeIndexes[AllConflictedDocsEtagsSlice]);
+
+            InsertIndexValuesFor(step, AttachmentsSchemaBase60, AttachmentsBucketAndEtagSlice,
+                fixedSizeIndex: AttachmentsSchemaBase60.FixedSizeIndexes[AttachmentsEtagSlice],
+                tableName: AttachmentsMetadataSlice.ToString());
+
+            return true;
+        }
+
+        private void InsertIndexValuesFor(UpdateStep step, TableSchema schema, Slice indexName, TableSchema.FixedSizeKeyIndexDef fixedSizeIndex, string tableName = null)
+        {
+            var fst = step.ReadTx.FixedTreeFor(fixedSizeIndex.Name, sizeof(long));
+            if (fst.NumberOfEntries == 0)
+                return;
+
+            var indexTree = step.WriteTx.CreateTree(indexName, isIndexTree: true);
+            var indexDef = schema.DynamicKeyIndexes[indexName];
+            
+            bool done = false;
+            long lastEtag = 0;
+
+            while (done == false)
+            {
+                var table = tableName == null
+                    ? new Table(schema, step.ReadTx)
+                    : step.ReadTx.OpenTable(schema, tableName);
+
+                var processedInCurrentTx = 0;
+
+                using (step.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                {
+                    context.TransactionMarkerOffset = (short)step.WriteTx.LowLevelTransaction.Id;
+
+                    var commit = false;
+                    var fromEtag = lastEtag == 0 ? 0 : lastEtag + 1;
+
+                    foreach (var tvh in table.SeekForwardFrom(fixedSizeIndex, fromEtag, 0))
+                    {
+                        if (processedInCurrentTx >= NumberOfItemsToMigrateInSingleTransaction ||
+                            _processedSinceLastCommit >= NumberOfItemsToMigrateInSingleTransaction)
+                        {
+                            commit = true;
+                            break;
+                        }
+
+                        var value = tvh.Reader;
+                        using (indexDef.GetValue(step.WriteTx.Allocator, ref value, out Slice val))
+                        {
+                            var index = new FixedSizeTree(step.WriteTx.LowLevelTransaction, indexTree, val, 0, isIndexTree: true);
+                            index.Add(value.Id);
+                        }
+
+                        processedInCurrentTx++;
+                        _processedSinceLastCommit++;
+                    }
+
+                    if (commit)
+                    {
+                        step.Commit(context);
+                        step.RenewTransactions();
+
+                        indexTree = step.WriteTx.ReadTree(indexName, isIndexTree: true);
+
+                        lastEtag += processedInCurrentTx;
```

Comment:
But `processedInCurrentTx` isn't the etag value. It's just an integer being incremented during the process.

Reply:
right, thanks
fixed